src/Ng/CommandHelpers.cs

```diff
@@ -127,7 +127,10 @@ public static StorageFactory CreateSuffixedStorageFactory(string suffix, IDictio
             return CreateStorageFactoryImpl(arguments, names, verbose);
         }
 
-        private static StorageFactory CreateStorageFactoryImpl(IDictionary<string, string> arguments, IDictionary<string, string> argumentNameMap, bool verbose)
+        private static StorageFactory CreateStorageFactoryImpl(IDictionary<string, string> arguments, 
+                                                               IDictionary<string, string> argumentNameMap,
+                                                               bool verbose,
+                                                               bool compressed = false)
```

Comment:
nit: did do we need optional parameter here? can we make the caller explicitly specify this in all cases? Given there have been multiple bugs in this area, i would prefer explicit over implicit